File: C++/Level1/소수 만들기.cpp
#include <vector>
#include <algorithm>
#include <iostream>
using namespace std;

bool isPrime(int num) {

    for (int i = 2; i <= num / 2; i++)
        if (num % i == 0)
            return false;

    return true;
}

int solution(vector<int> nums) {
    
    int answer = 0;
    int len = nums.size();

    vector<int> mask(len);
    for (int i = len - 3; i < len; i++)
        mask[i] = 1;

    do {
        int sum = 0;
        for (int i = 0; i < len; i++)
            if (mask[i])
                sum += nums[i];

        if (isPrime(sum))
            answer++;

    } while (next_permutation(mask.begin(), mask.end()));

    return answer;
}

File: programmers-coding-test/Java/Level1/소수 만들기.java
class Solution {
    public int solution(int[] nums) {
        int answer = 0;
        int len = nums.length;

        for (int i = 0; i < len - 2; i++)
            for (int j = i + 1; j < len - 1; j++)
                for (int k = j + 1; k < len; k++)
                    if (isPrime(nums[i] + nums[j] + nums[k]))
                        answer++;

        return answer;
    }

    public boolean isPrime(int num) {
        for (int i = 2; i <= num / 2; i++)
            if (num % i == 0)
                return false;
        return true;
    }
}